### PR TITLE
Small performance improvements

### PR DESCRIPTION
**Pull request by aanm**
```release-note
Small performance improvements to decrease policy regeneration time
```

Please see per commit basis

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5269)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

File: pkg/endpoint/policy.go [hunk 1]
```diff
@@ -529,6 +529,12 @@ func (e *Endpoint) regeneratePolicy(owner Owner, opts models.ConfigurationMap) (
 		labelsMap = e.prevIdentityCache
 	}
 
+	// Containers without a security identity are not accessible
+	if e.SecurityIdentity == nil {
+		e.getLogger().Warn("Endpoint lacks identity, skipping policy calculation")
+		return false, nil
+	}
+
```
Comment (ianvernon) [inline pkg/endpoint/policy.go:537]: Can you confirm moving this code here is safe to do before evaluating the logic on lines 545-554? I.e., is the logic: 
```
!e.forcePolicyCompute && e.nextPolicyRevision >= revision && labelsMap == e.prevIdentityCache && opts == nil
```

always guaranteed to evaluate to `false` if `e.SecurityIdentity == nil`?

Comment (rlenglet) [reply pkg/endpoint/policy.go:537]: Does it matter? If `e.SecurityIdentity == nil`, we'll generate a `DROP_ALL` BPF program, so it doesn't seem like any of those parameters (revision, ID cache, config) matter?

**Review by rlenglet — APPROVED** [review]

File: pkg/endpoint/policy.go (hunk 1 above)
Comment (ianvernon) [reply pkg/endpoint/policy.go:537]: Makes sense.

**Review by ianvernon — APPROVED** [review]

**Pull request merged**